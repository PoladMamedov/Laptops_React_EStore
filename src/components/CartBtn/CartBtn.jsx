export function CartBtn(props) {
  return (
    <button data-testid="cart-btn" type="button" className="btn btn-light border-secondary position-relative pt-0">
      <img width={20} height={20} src="./img/shopping-cart.png" alt="cart" />
      <span className="position-absolute top-0 start-100 translate-middle badge rounded-pill bg-danger">
        {props.cartItemsCount}
      </span>
    </button>
  );
}