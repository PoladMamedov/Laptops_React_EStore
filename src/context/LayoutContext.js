import React, { useContext, useState } from "react";
const LayoutContext = React.createContext();
const LayoutContextUpdate = React.createContext();

export function useLayout() {
  return useContext(LayoutContext);
}
export function useLayoutUpdate() {
  return useContext(LayoutContextUpdate);
}

export function LayoutProvider({ children }) {
  const [isListLayout, setIsListLayout] = useState(false);

  function setCardLayout() {
    setIsListLayout(false);
  }
  function setListLayout() {
    setIsListLayout(true);
  }

  return (
    <LayoutContext.Provider value={isListLayout}>
      <LayoutContextUpdate.Provider value={{ setListLayout, setCardLayout }}>{children}</LayoutContextUpdate.Provider>
    </LayoutContext.Provider>
  );
}
