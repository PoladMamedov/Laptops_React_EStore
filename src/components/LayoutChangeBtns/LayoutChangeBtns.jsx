import { useEffect } from "react";
import { useLayout, useLayoutUpdate } from "../../context/LayoutContext";
import { useWindowWidth } from "../../hooks/useWindowWidth";

export function LayoutChangeBtns() {
  const isListlayout = useLayout();
  const { setListLayout, setCardLayout } = useLayoutUpdate();
  const windowWidth = useWindowWidth();
  useEffect(() => {
    if (localStorage.getItem("isListLayout") === "false") {
      localStorage.setItem("islistLayout", "false");
      setCardLayout();
    }
    if (localStorage.getItem("isListLayout") === "true") {
      localStorage.setItem("islistLayout", "true");
      setListLayout();
    }
    if (windowWidth < 580) setCardLayout();
  });
  return windowWidth < 580 ? null : (
    <div data-testid="layout-btns" className="btn-group">
      <input type="checkbox" className="btn-check" checked={isListlayout ? false : true} readOnly />
      <label
        onClick={() => {
          localStorage.setItem("isListLayout", "false");
          setCardLayout();
        }}
        className="btn btn-outline-secondary"
      >
        Cards
      </label>
      <input type="checkbox" className="btn-check" checked={isListlayout ? true : false} readOnly />
      <label
        onClick={() => {
          localStorage.setItem("isListLayout", "true");
          setListLayout();
        }}
        className="btn btn-outline-secondary"
      >
        List
      </label>
    </div>
  );
}
