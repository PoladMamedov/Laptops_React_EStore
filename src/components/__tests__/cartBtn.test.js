import "@testing-library/jest-dom";
import { render, screen, cleanup } from "@testing-library/react";
import { CartBtn } from "../CartBtn/CartBtn";
import { LayoutChangeBtns } from "../LayoutChangeBtns/LayoutChangeBtns";
import { LayoutProvider } from "../../context/LayoutContext";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("should render cart button without text content", () => {
  render(<CartBtn cartItemsCount={0} />);
  expect(screen.getByTestId("cart-btn")).toBeInTheDocument();
  expect(screen.getByTestId("cart-btn")).toHaveTextContent("0");
});
test("should match the snapshot", () => {
  const cartBtnSnap = renderer.create(<CartBtn cartItemsCount={0} />);
  expect(cartBtnSnap).toMatchSnapshot();
});

test("should render proper layout buttons", () => {
  render(
    <LayoutProvider>
      <LayoutChangeBtns />
    </LayoutProvider>
  );

  expect(screen.getByTestId("layout-btns")).toBeInTheDocument();
  expect(screen.getByTestId("layout-btns")).toHaveTextContent("Card");
  expect(screen.getByTestId("layout-btns")).toHaveTextContent("List");
});
