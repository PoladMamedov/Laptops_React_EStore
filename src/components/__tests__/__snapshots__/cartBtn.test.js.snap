// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<button
  className="btn btn-light border-secondary position-relative pt-0"
  data-testid="cart-btn"
  type="button"
>
  <img
    alt="cart"
    height={20}
    src="./img/shopping-cart.png"
    width={20}
  />
  <span
    className="position-absolute top-0 start-100 translate-middle badge rounded-pill bg-danger"
  >
    0
  </span>
</button>
`;
